spark/src/main/java/io/spark/ddf/etl/SqlHandler.java

```diff
@@ -109,12 +115,19 @@ public DDF sql2ddf(String command, Schema schema, String dataSource, DataFormat
   @Override
   public List<String> sql2txt(String command, Integer maxRows, String dataSource) throws DDFException {
     DataFrame rdd = ((SparkDDFManager) this.getManager()).getHiveContext().sql(command);
+
+    /*
     Row[] arrRow = rdd.collect();
     List<String> lsString = new ArrayList<String>();
 
     for (Row row : arrRow) {
       lsString.add(row.mkString("\t"));
     }
-    return lsString;
+    return lsString
+    */
+
+    String[] strResult = SparkUtils.jsonForComplexType(rdd, "\t");
```

Comment:
use DataFrame.toJson to convert the DataFrame to RDD[String]
https://github.com/apache/spark/blob/branch-1.3/sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala#L1236

@namma